### PR TITLE
MUMUP-2260 : Taco, err um, i mean Bucky Announcements

### PR DESCRIPTION
**Pull request by timlevett**
+ Adds creeper state, :tm:, when you have unread announcements

image

+ On hover reveals bucky with short title and description

image

+ On click on bucky he hides again
+ On click of any header link or the "tell me more" it brings you to features and updates your "last read"
+ On click of "not interested" it just updates your "last read"
+ Multiple Announcement support (up to 3)

image

_We have 4 new ones here, but we only see the latest 3_

+ Only show announcements that are not expired, if something is expired update it to latest read
+ Mobile friendly mode

image

+ GA this. `category `: feature ; `actions` : hide-bucky, show-more, dismiss; `label` : last id or `na`

### Related changes
+ Refactors popup functionality to work with this. Popup just updates to last popup, announcements update just announcements. Shouldn't see any difference in popup functionality.
+ Move PopupController and partial for modal to the feature section of code. Made more sense there.

### CODE REVIEW

**Comment by keirserrie** [comment]
Excited to see this :+1:

**Comment by vertein** [comment]
:+1:

**Comment by jhanstra** [comment]
Awesome work, :+1:

File: src/main/webapp/js/app-config.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ define(['angular'], function(angular) {
     var config = angular.module('app-config', []);
     config
         .constant('APP_FLAGS', {
-            'features' : false,
+            'features' : true,
```
Comment (timlevett) [inline src/main/webapp/js/app-config.js]: crap, meant to switch this back to false. brb

**Comment by keirserrie** [comment]
Great! Is it possilble to have Bucky go all the way to the top & have the "speech bubble" styling there?

If not, then still :+1: 

screenshot 2015-12-02 15 38 51

**Comment by nblair** [comment]
Bucky winks! This is great!

**Comment by timlevett** [comment]
@keirserrie : I can make him go all the way up, forgot about that. The speech bubble might be tough as i'm just doing a border around the whole thing right now.

**Comment by smargovsky** [comment]
:+1:  for feature, and :100: for creeper state (:tm:)

**Comment by smargovsky** [comment]
@timlevett : Regarding @keirserrie's comment about the arrow - what about using Angular UI Bootstrap popover w/ custom css for the arrow ( like http://www.bootply.com/88325 )

**Comment by timlevett** [comment]
Sweet @smargovsky . I'll check it out. Thanks!

**Comment by timlevett** [comment]
Took advice from @keirserrie and added in the arrow with help from @smargovsky . Thanks!

image

Final product. Note I added notification bell just to make sure all was well when that was enabled.

**Comment by jhanstra** [comment]
:+1: again :100:

**Pull request merged**